src/pb.rs

```diff
@@ -42,11 +43,10 @@ pub enum Units {
     Bytes,
 }
 
-#[derive(Debug)]
-pub struct ProgressBar {
+pub struct ProgressBar<T: Write> {
     start_time: SteadyTime,
     units: Units,
-    total: u64,
+    pub total: u64,
```

Comment:
For certain tasks, specially those that are parallelized, the total is not known when instantiating the bar. 

Take for example a script that walks a directory structure and does an asynchronous operation on every file that it finds. This allows a client to update the total number of bytes to be processed as it discovers new files.

I think it should be exposed as a method, which would help with backwards compatibility if in the future we need do something else when the client updates the total. WDYT?